### PR TITLE
Update Service interface to use Reconcile/Delete methods

### PR DESCRIPTION
**Pull request by justaugustus**
Signed-off-by: Stephen Augustus <saugustus@vmware.com>

**What this PR does / why we need it**:
- Update Makefile
- s/`CreateOrUpdate`/`Reconcile`
- Add `GetterService` interface, a temporary interface to represent services that still require Get methods
- Add type conversions (`var _ azure.Service = (*Service)(nil)`) to each Azure service, which will force compilation errors if an Azure service doesn't satisfy the `Service` interface

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related: #110, #174, #177 

**Special notes for your reviewer**:

This is largely a search and replace, so no additional tests are required.
As we refactor the individual lower-level reconcilers, we'll add unit tests alongside.

**Release note**:

```release-note
Update Service interface to use Reconcile/Delete methods
```

/milestone v1alpha1
/priority important-soon
/kind design

### CODE REVIEW

**Comment by justaugustus** [comment]
Ready for review.
/assign @awesomenix @soggiest

File: pkg/cloud/azure/services/groups/groups.go [hunk 1]
```diff
@@ -31,8 +31,8 @@ func (s *Service) Get(ctx context.Context, spec azure.Spec) (interface{}, error)
 	return s.Client.Get(ctx, s.Scope.ClusterConfig.ResourceGroup)
 }
 
-// CreateOrUpdate creates or updates a resource group.
-func (s *Service) CreateOrUpdate(ctx context.Context, spec azure.Spec) error {
+// // Reconcile gets/creates/updates a resource group.
+func (s *Service) Reconcile(ctx context.Context, spec azure.Spec) error {
 	klog.V(2).Infof("creating resource group %s", s.Scope.ClusterConfig.ResourceGroup)
```
Comment (awesomenix) [inline pkg/cloud/azure/services/groups/groups.go]: No call to get? Reconcile is 

if resourcegroup.Get() ; Is Expected==Current {
return
} else {
createorupdate()
}

isnt it?

File: pkg/cloud/azure/actuators/cluster/actuator_test.go [hunk 2]
```diff
@@ -157,27 +157,35 @@ func TestServicesCreatedCount(t *testing.T) {
 	if cache[azure.GenerateVnetName(fakeReconciler.scope.Cluster.Name)] != 1 {
 		t.Errorf("Expected 1 count of %s service", azure.GenerateVnetName(fakeReconciler.scope.Cluster.Name))
 	}
+
```
Comment (soggiest) [inline pkg/cloud/azure/actuators/cluster/actuator_test.go]: what are all these pluses?

File: pkg/cloud/azure/services/groups/groups.go (hunk 1 above)
Comment (awesomenix) [reply pkg/cloud/azure/services/groups/groups.go]: same comments for other services

File: pkg/cloud/azure/actuators/cluster/actuator_test.go (hunk 2 above)
Comment (justaugustus) [reply pkg/cloud/azure/actuators/cluster/actuator_test.go]: just adding spaces between the `if` statements.

File: pkg/cloud/azure/actuators/machine/actuator_test.go [hunk 3]
```diff
@@ -203,8 +205,8 @@ func (s *FakeVMService) Get(ctx context.Context, spec azure.Spec) (interface{},
 	}, nil
 }
 
-// CreateOrUpdate returns fake success.
-func (s *FakeVMService) CreateOrUpdate(ctx context.Context, spec azure.Spec) error {
+// Reconcile returns fake success.
+func (s *FakeVMService) Reconcile(ctx context.Context, spec azure.Spec) error {
```
Comment (awesomenix) [inline pkg/cloud/azure/actuators/machine/actuator_test.go]: Please add test cases

1. Get is called
2. CreateOrUpdate is called only when expected != current 

Rename the fields in test cases here

**Pull request merged**